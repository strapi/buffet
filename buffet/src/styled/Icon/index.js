/**
 *
 * Icon
 *
 */

import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import colors from '../../assets/styles/colors';
import sizes from '../../assets/styles/sizes';

const Icon = styled.span`
  width: ${sizes.input.height};
  height: ${sizes.input.height};
  display: flex;
  ::before {
    font-family: 'FontAwesome';
    font-weight: 600;
    font-size: 14px;
    line-height: 100%;
    display: inline-block;
    vertical-align: middle;
    margin: auto;
    color: ${colors.greyIconColor};
  }
    
  /*
   * Specific icon
   */

  // Search
  ${props =>
    props.type === 'search' &&
    css`
      background-color: ${colors.greyIconBkgd};
      ::before {
        content: '\f002';
      }
    `}}

  // Email
  ${props =>
    props.type === 'email' &&
    css`
      background-color: ${colors.greyIconBkgd};
      ::before {
        content: '\f1fa';
      }
    `}}
    
  // Password
  ${props =>
    props.type === 'password' &&
    css`
      ::before {
        content: '\f06e';
      }
    `}}
`;

Icon.defaultProps = {
  type: 'search',
};

Icon.propTypes = {
  type: PropTypes.string,
};

export default Icon;
