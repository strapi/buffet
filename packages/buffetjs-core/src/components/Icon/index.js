import React, { isValidElement } from 'react';
import PropTypes from 'prop-types';
import {
  faSearch,
  faEye,
  faAngleLeft,
  faAngleRight,
  faSortDown,
  faSortUp,
  faTrash,
  faPlus,
} from '@fortawesome/free-solid-svg-icons';
import { faClock } from '@fortawesome/free-regular-svg-icons';
import { Icon as StyledIcon } from '@buffetjs/styles';

const iconMap = new Map([
  ['time', faClock],
  ['password', faEye],
  ['search', faSearch],
  ['left', faAngleLeft],
  ['right', faAngleRight],
  ['asc', faSortUp],
  ['desc', faSortDown],
  ['trash', faTrash],
  ['plus', faPlus],
]);

function Icon({ icon, className }) {
  if (iconMap.has(icon)) {
    return (
      <StyledIcon icon={iconMap.get(icon)} className={className || undefined} />
    );
  }
  if (isValidElement(icon)) {
    return <span className={className}>{icon}</span>;
  }

  return null;
}

Icon.defaultProps = {
  className: null,
  icon: 'search',
};

Icon.propTypes = {
  className: PropTypes.string,
  icon: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
};

export default Icon;
