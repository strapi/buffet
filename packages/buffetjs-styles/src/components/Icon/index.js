/**
 *
 * Icon
 *
 */

import styled from 'styled-components';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Icon = styled(FontAwesomeIcon)``;

Icon.propTypes = {
  icon: PropTypes.any.isRequired,
};

export default Icon;
