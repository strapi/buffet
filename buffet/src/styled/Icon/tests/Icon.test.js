import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Icon from '../index';

let renderedComponent;
const renderComponent = (props = {}) => mount(<Icon {...props} />);

describe('<Icon />', () => {
  describe('<Icon /> attributes', () => {
    afterEach(() => {
      renderedComponent.unmount();
    });

    it('should have a default type equals to search', () => {
      renderedComponent = renderComponent({ type: 'search' });
      const defaultType = 'search';
      expect(renderedComponent.at(0).prop('type')).toBe(defaultType);
    });
  });
  describe('<Icon /> icon attributes', () => {
    it('should render the search icon', () => {
      const tree = renderer.create(<Icon type="search" />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render the email icon', () => {
      const tree = renderer.create(<Icon type="email" />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render the password icon', () => {
      const tree = renderer.create(<Icon type="password" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
