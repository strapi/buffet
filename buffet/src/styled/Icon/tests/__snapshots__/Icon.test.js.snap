// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> <Icon /> icon attributes should render the email icon 1`] = `
<span
  className="sc-bdVaJa fBhdmK"
  type="email"
/>
`;

exports[`<Icon /> <Icon /> icon attributes should render the password icon 1`] = `
<span
  className="sc-bdVaJa jSlhwP"
  type="password"
/>
`;

exports[`<Icon /> <Icon /> icon attributes should render the search icon 1`] = `
<span
  className="sc-bdVaJa jmujDp"
  type="search"
/>
`;
